keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/BetaVertex.java

```diff
@@ -21,10 +25,12 @@
 import static io.improbable.keanu.tensor.TensorShapeValidation.checkHasSingleNonScalarShapeOrAllScalar;
 import static io.improbable.keanu.tensor.TensorShapeValidation.checkTensorsMatchNonScalarShapeOrAreScalar;
 
-public class BetaVertex extends DoubleVertex implements Differentiable, ProbabilisticDouble, SamplableWithManyScalars<DoubleTensor> {
+public class BetaVertex extends DoubleVertex implements SaveableVertex, Differentiable, ProbabilisticDouble, SamplableWithManyScalars<DoubleTensor> {
 
     private final DoubleVertex alpha;
     private final DoubleVertex beta;
+    private static final String ALPHA_NAME = "alpha";
```

Comment:
you've introduced lots of `"alpha"`s etc. - these should be consolidated in one place, and preferably consolidated with the existing `ParameterName`s in `io.improbable.keanu.distributions.hyperparam.Diffs` (which could be moved somewhere else, since they have nothing to do with differentiation specifically)

Reply:
I think I've gone off the ParameterName approach. I think it unnecessarily tightly couples all of the distribution vertices. There might be two distribution that have a hyper param called alpha but I think the duplicated code that defines it's name (e.g. `private static final String ALPHA_NAME = "alpha";`) is worth it in order to avoid the tight coupling that removing the duplication would cause.

Reply:
I'm not particularly attached to the `ParameterNames` but don't you think some consistency with the strings is still important? Not just for the protobufs but for text-based serialisation formats too e.g. for the DOT format.